datatable/java/datatable/src/main/java/io/cucumber/datatable/TablePrinter.java

```diff
@@ -2,30 +2,46 @@
 
 import java.io.IOException;
 import java.util.List;
+import org.apiguardian.api.API;
 
-class TablePrinter {
+@API(status = API.Status.STABLE)
+public class TablePrinter {
     private int[][] cellLengths;
     private int[] maxLengths;
 
-    void printTable(List<List<String>> table, StringBuilder appendable) {
+    private String startIndent = "      ";
+    private Boolean escapeCells = true;
+
+
+    public TablePrinter withIndent(String startIndent) {
+        this.startIndent = startIndent;
+        return this;
+    }
+
+    public TablePrinter escapeCells(Boolean escapeCells) {
+        this.escapeCells = escapeCells;
+        return this;
+    }
+
+    public void printTable(DataTable table, StringBuilder appendable) {
```

Comment:
wasn't sure if your preference was to pass the table in the method, or when first calling the `TablePrinter()` constructor. I figured it's more likely this could be used to print multiple different tables with the same print configuration, so best to keep it as a method argument.

Reply:
:+1:

Reply:
You've got it.